core/domain/stats_jobs_one_off.py

```diff
@@ -18,14 +18,403 @@
+            exp_id, latest_exp_version, num_starts, 0, num_actual_starts, 0,
+            num_completions, 0, state_stats_mapping)
+        stats_services.create_stats_model(exploration_stats)
+
+
+class StatisticsAuditV2(jobs.BaseMapReduceOneOffJobManager):
+    """A one-off statistics audit.
+
+    Performs a brief audit of exploration stats values to ensure that they
+    maintain simple sanity checks like being non-negative and so on.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [stats_models.ExplorationStatsModel]
+
+    @classmethod
+    def require_non_negative(
+            cls, exp_id, property_name, value, state_name=None):
+        if not state_name:
+            if value[property_name] < 0:
+                yield (
+                    'Negative %s count: exp_id:%s version:%s %s:%s' % (
+                        property_name, exp_id, value['version'], property_name,
+                        value[property_name]),)
+        else:
+            if value['state_stats_mapping'][state_name][property_name] < 0:
+                yield (
+                    'Negative %s count: exp_id:%s version:%s state:%s %s:%s' % (
+                        property_name, exp_id, value['version'], state_name,
+                        property_name,
+                        value['state_stats_mapping'][state_name][
+                            property_name]),)
+
+    @staticmethod
+    def map(item):
+        """Implements the map function. Must be declared @staticmethod.
+
+        Args:
+            item. ExplorationStatsModel.
+
+        Yields:
+            tuple. For ExplorationStatsModel, a 2-tuple of the form
+                (exp_id, value) where value is of the form:
+                    {
+                        'version': int. Version of the exploration.
+                        'num_starts': int. # of times exploration was started.
+                        'num_completions': int. # of times exploration was
+                            completed.
+                        'num_actual_starts': int. # of times exploration was
+                            actually started.
+                        'state_stats_mapping': A dict containing the values of
+                            stats for the states of the exploration. It is
+                            formatted as follows:
+                            {
+                                state_name: {
+                                    'total_answers_count',
+                                    'useful_feedback_count',
+                                    'total_hit_count',
+                                    'first_hit_count',
+                                    'num_completions'
+                                }
+                            }
+                    }
+        """
+        reduced_state_stats_mapping = {
+            state_name: {
+                'total_answers_count': item.state_stats_mapping[state_name][
```

Comment:
Keep _v1 in the keys. total_answers_count means something different (it means the aggregation of v1 and subsequent versions).